### PR TITLE
Improve #51

### PR DESCRIPTION
**Pull request by meyskens**
Improvements done to the Docker setup proposed in #51

### CODE REVIEW

**Comment by lily-mayfield** [comment]
@meyskens 
```
(api) lily@ThinkPad-S1-Yoga:~/OSP/repos/osp-api (meyskens-feature/make-deployable)*$ docker-compose -f docker/docker-compose.test.yml build
ERROR: Cannot extend service 'api' in /home/lily/OSP/repos/osp-api/docker/docker-compose.base.yml: services with 'links' cannot be extended
```

File: Dockerfile [hunk 1]
```diff
@@ -1 +1,6 @@
-FROM python:3.5-onbuild
+FROM python:3.5-alpine
```
Comment (meyskens) [inline Dockerfile]: Shouldn't we use Python 2.7, the real one? (just kidding :laughing:)

**Comment by lily-mayfield** [comment]
Still getting the same error as I posted above.

File: docker-compose.base.yml [hunk 2]
```diff
@@ -6,26 +6,21 @@ services:
...
             context: .
-            dockerfile: Dockerfile-api
         restart: always
-        entrypoint: ./api-first-run.sh
-        command: gunicorn --config=gunicorn.py --reload osp_api.wsgi:app
         env_file:
             - api-variables.env
-        volumes:
-            - .:/osp_api
+        links:
```
Comment (lily-mayfield) [inline docker-compose.base.yml]: i think, annoyingly, this can't be in base... it must be defined in everything which inherits... know any work arounds?

Comment (meyskens) [inline docker-compose.base.yml]: define it in the inherits :p there is an issue filed I see but never been done something about

**Comment by lily-mayfield** [comment]
I'm getting requirements-related issues now.

**Comment by meyskens** [comment]
@lily-mayfield logs?

**Comment by lily-mayfield** [comment]
@meyskens 

```
api_1       | Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-1vpxicjz/psycopg2/
api_1       | Traceback (most recent call last):
api_1       |   File "manage.py", line 8, in <module>
api_1       |     from flask_script import Manager
api_1       | ImportError: No module named 'flask_script'
api_1       | Traceback (most recent call last):
api_1       |   File "manage.py", line 8, in <module>
api_1       |     from flask_script import Manager
api_1       | ImportError: No module named 'flask_script'
api_1       | Traceback (most recent call last):
api_1       |   File "manage.py", line 8, in <module>
api_1       |     from flask_script import Manager
api_1       | ImportError: No module named 'flask_script'
api_1       | /bin/sh: gunicorn: not found
api_1       | Collecting alabaster==0.7.10 (from -r /osp_api/requirements.txt (line 1))
api_1       |   Using cached alabaster-0.7.10-py2.py3-none-any.whl
api_1       | Collecting alembic==0.9.3 (from -r /osp_api/requirements.txt (line 2))
api_1       |   Using cached alembic-0.9.3.tar.gz
api_1       | Collecting aniso8601==1.2.1 (from -r /osp_api/requirements.txt (line 3))
api_1       |   Using cached aniso8601-1.2.1.tar.gz
api_1       | Collecting apipkg==1.4 (from -r /osp_api/requirements.txt (line 4))
api_1       |   Using cached apipkg-1.4-py2.py3-none-any.whl
api_1       | Collecting Babel==2.4.0 (from -r /osp_api/requirements.txt (line 5))
api_1       |   Using cached Babel-2.4.0-py2.py3-none-any.whl
api_1       | Collecting blinker==1.4 (from -r /osp_api/requirements.txt (line 6))
api_1       |   Using cached blinker-1.4.tar.gz
api_1       | Collecting certifi==2017.4.17 (from -r /osp_api/requirements.txt (line 7))
api_1       |   Using cached certifi-2017.4.17-py2.py3-none-any.whl
api_1       | Collecting chardet==3.0.4 (from -r /osp_api/requirements.txt (line 8))
api_1       |   Using cached chardet-3.0.4-py2.py3-none-any.whl
api_1       | Collecting click==6.7 (from -r /osp_api/requirements.txt (line 9))
api_1       |   Using cached click-6.7-py2.py3-none-any.whl
api_1       | Collecting coverage==4.4.1 (from -r /osp_api/requirements.txt (line 10))
api_1       |   Using cached coverage-4.4.1.tar.gz
api_1       | Collecting coveralls==1.1 (from -r /osp_api/requirements.txt (line 11))
api_1       |   Using cached coveralls-1.1-py2.py3-none-any.whl
api_1       | Collecting docopt==0.6.2 (from -r /osp_api/requirements.txt (line 12))
api_1       |   Using cached docopt-0.6.2.tar.gz
api_1       | Collecting docutils==0.13.1 (from -r /osp_api/requirements.txt (line 13))
api_1       |   Using cached docutils-0.13.1-py3-none-any.whl
api_1       | Collecting execnet==1.4.1 (from -r /osp_api/requirements.txt (line 14))
api_1       |   Using cached execnet-1.4.1-py2.py3-none-any.whl
api_1       | Collecting Flask==0.12.2 (from -r /osp_api/requirements.txt (line 15))
api_1       |   Using cached Flask-0.12.2-py2.py3-none-any.whl
api_1       | Collecting Flask-BabelEx==0.9.3 (from -r /osp_api/requirements.txt (line 16))
api_1       |   Using cached Flask-BabelEx-0.9.3.tar.gz
api_1       | Collecting Flask-Caching==1.3.2 (from -r /osp_api/requirements.txt (line 17))
api_1       |   Using cached Flask_Caching-1.3.2-py2.py3-none-any.whl
api_1       | Collecting Flask-JWT-Extended==3.1.1 (from -r /osp_api/requirements.txt (line 18))
api_1       |   Using cached Flask-JWT-Extended-3.1.1.tar.gz
api_1       | Collecting Flask-Limiter==0.9.4 (from -r /osp_api/requirements.txt (line 19))
api_1       |   Using cached Flask-Limiter-0.9.4.tar.gz
api_1       | Collecting Flask-Login==0.4.0 (from -r /osp_api/requirements.txt (line 20))
api_1       |   Using cached Flask_Login-0.4.0-py2.py3-none-any.whl
api_1       | Collecting Flask-Mail==0.9.1 (from -r /osp_api/requirements.txt (line 21))
api_1       |   Using cached Flask-Mail-0.9.1.tar.gz
api_1       | Collecting Flask-Migrate==2.0.4 (from -r /osp_api/requirements.txt (line 22))
api_1       |   Using cached Flask-Migrate-2.0.4.tar.gz
api_1       | Collecting Flask-Principal==0.4.0 (from -r /osp_api/requirements.txt (line 23))
api_1       |   Using cached Flask-Principal-0.4.0.tar.gz
api_1       | Collecting Flask-RESTful==0.3.6 (from -r /osp_api/requirements.txt (line 24))
api_1       |   Using cached Flask_RESTful-0.3.6-py2.py3-none-any.whl
api_1       | Collecting Flask-Script==2.0.5 (from -r /osp_api/requirements.txt (line 25))
api_1       |   Using cached Flask-Script-2.0.5.tar.gz
api_1       | Collecting Flask-Security==3.0.0 (from -r /osp_api/requirements.txt (line 26))
api_1       |   Using cached Flask_Security-3.0.0-py2.py3-none-any.whl
api_1       | Collecting Flask-SQLAlchemy==2.2 (from -r /osp_api/requirements.txt (line 27))
api_1       |   Using cached Flask_SQLAlchemy-2.2-py2.py3-none-any.whl
api_1       | Collecting Flask-WTF==0.14.2 (from -r /osp_api/requirements.txt (line 28))
api_1       |   Using cached Flask_WTF-0.14.2-py2.py3-none-any.whl
api_1       | Collecting gunicorn==19.7.1 (from -r /osp_api/requirements.txt (line 29))
api_1       |   Using cached gunicorn-19.7.1-py2.py3-none-any.whl
api_1       | Collecting idna==2.5 (from -r /osp_api/requirements.txt (line 30))
api_1       |   Using cached idna-2.5-py2.py3-none-any.whl
api_1       | Collecting imagesize==0.7.1 (from -r /osp_api/requirements.txt (line 31))
api_1       |   Using cached imagesize-0.7.1-py2.py3-none-any.whl
api_1       | Collecting itsdangerous==0.24 (from -r /osp_api/requirements.txt (line 32))
api_1       |   Using cached itsdangerous-0.24.tar.gz
api_1       | Collecting Jinja2==2.9.6 (from -r /osp_api/requirements.txt (line 33))
api_1       |   Using cached Jinja2-2.9.6-py2.py3-none-any.whl
api_1       | Collecting limits==1.2.1 (from -r /osp_api/requirements.txt (line 34))
api_1       |   Using cached limits-1.2.1.tar.gz
api_1       | Collecting Mako==1.0.7 (from -r /osp_api/requirements.txt (line 35))
api_1       |   Using cached Mako-1.0.7.tar.gz
api_1       | Collecting MarkupSafe==1.0 (from -r /osp_api/requirements.txt (line 36))
api_1       |   Using cached MarkupSafe-1.0.tar.gz
api_1       | Collecting marshmallow==2.13.5 (from -r /osp_api/requirements.txt (line 37))
api_1       |   Using cached marshmallow-2.13.5-py2.py3-none-any.whl
api_1       | Collecting passlib==1.7.1 (from -r /osp_api/requirements.txt (line 38))
api_1       |   Using cached passlib-1.7.1-py2.py3-none-any.whl
api_1       | Collecting pockets==0.5.1 (from -r /osp_api/requirements.txt (line 39))
api_1       |   Using cached pockets-0.5.1-py2.py3-none-any.whl
api_1       | Collecting psycopg2==2.7.1 (from -r /osp_api/requirements.txt (line 40))
api_1       |   Using cached psycopg2-2.7.1.tar.gz
api_1       |     Complete output from command python setup.py egg_info:
api_1       |     running egg_info
api_1       |     creating pip-egg-info/psycopg2.egg-info
api_1       |     writing dependency_links to pip-egg-info/psycopg2.egg-info/dependency_links.txt
api_1       |     writing top-level names to pip-egg-info/psycopg2.egg-info/top_level.txt
api_1       |     writing pip-egg-info/psycopg2.egg-info/PKG-INFO
api_1       |     writing manifest file 'pip-egg-info/psycopg2.egg-info/SOURCES.txt'
api_1       |     Error: pg_config executable not found.
api_1       |     
api_1       |     Please add the directory containing pg_config to the PATH
api_1       |     or specify the full executable path with the option:
api_1       |     
api_1       |         python setup.py build_ext --pg-config /path/to/pg_config build ...
api_1       |     
api_1       |     or with the pg_config option in 'setup.cfg'.
api_1       |     
api_1       |     ----------------------------------------
api_1       | Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-qptoz9vg/psycopg2/
api_1       | Traceback (most recent call last):
api_1       |   File "manage.py", line 8, in <module>
api_1       |     from flask_script import Manager
api_1       | ImportError: No module named 'flask_script'
api_1       | Traceback (most recent call last):
api_1       |   File "manage.py", line 8, in <module>
api_1       |     from flask_script import Manager
api_1       | ImportError: No module named 'flask_script'
api_1       | Traceback (most recent call last):
api_1       |   File "manage.py", line 8, in <module>
api_1       |     from flask_script import Manager
api_1       | ImportError: No module named 'flask_script'
api_1       | /bin/sh: gunicorn: not found
ospapi_api_1 exited with code 127

```

File: api-first-run.sh [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
+pip install -U -r /osp_api/requirements.txt
```
Comment (lily-mayfield) [inline api-first-run.sh]: this doesn't fix it; i tried that

Comment (meyskens) [inline api-first-run.sh]: trying out atm, pulling images but delayed as i'm updating my images

Comment (meyskens) [inline api-first-run.sh]: is the error on run?

Comment (lily-mayfield) [inline api-first-run.sh]: same error. all of the requirements are in the requires

**Comment by lily-mayfield** [comment]
@meyskens This is broken too. What were the big changes? Maybe I can just apply them myself on the version I have stashed. To remove `expose` and use `link`... and possibly to use alpine? any other big changes?

**Comment by meyskens** [comment]
@lily-mayfield all should be in the commit logs ;) what is the error?

**Comment by lily-mayfield** [comment]
```
api_1       | Traceback (most recent call last):
api_1       |   File "/usr/local/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 2147, in _wrap_pool_connect
api_1       |     return fn()
api_1       |   File "/usr/local/lib/python3.5/site-packages/sqlalchemy/pool.py", line 387, in connect
api_1       |     return _ConnectionFairy._checkout(self)
api_1       |   File "/usr/local/lib/python3.5/site-packages/sqlalchemy/pool.py", line 766, in _checkout
api_1       |     fairy = _ConnectionRecord.checkout(pool)
api_1       |   File "/usr/local/lib/python3.5/site-packages/sqlalchemy/pool.py", line 516, in checkout
api_1       |     rec = pool._do_get()
api_1       |   File "/usr/local/lib/python3.5/site-packages/sqlalchemy/pool.py", line 1138, in _do_get
api_1       |     self._dec_overflow()
api_1       |   File "/usr/local/lib/python3.5/site-packages/sqlalchemy/util/langhelpers.py", line 66, in __exit__
api_1       |     compat.reraise(exc_type, exc_value, exc_tb)
api_1       |   File "/usr/local/lib/python3.5/site-packages/sqlalchemy/util/compat.py", line 187, in reraise
api_1       |     raise value
api_1       |   File "/usr/local/lib/python3.5/site-packages/sqlalchemy/pool.py", line 1135, in _do_get
api_1       |     return self._create_connection()
api_1       |   File "/usr/local/lib/python3.5/site-packages/sqlalchemy/pool.py", line 333, in _create_connection
api_1       |     return _ConnectionRecord(self)
api_1       |   File "/usr/local/lib/python3.5/site-packages/sqlalchemy/pool.py", line 461, in __init__
api_1       |     self.__connect(first_connect_check=True)
api_1       |   File "/usr/local/lib/python3.5/site-packages/sqlalchemy/pool.py", line 651, in __connect
api_1       |     connection = pool._invoke_creator(self)
api_1       |   File "/usr/local/lib/python3.5/site-packages/sqlalchemy/engine/strategies.py", line 105, in connect
api_1       |     return dialect.connect(*cargs, **cparams)
api_1       |   File "/usr/local/lib/python3.5/site-packages/sqlalchemy/engine/default.py", line 393, in connect
api_1       |     return self.dbapi.connect(*cargs, **cparams)
api_1       |   File "/usr/local/lib/python3.5/site-packages/psycopg2/__init__.py", line 130, in connect
api_1       |     conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
api_1       | psycopg2.OperationalError: could not connect to server: Connection refused
api_1       | 	Is the server running on host "postgres" (172.18.0.2) and accepting
api_1       | 	TCP/IP connections on port 5432?
```

**Comment by meyskens** [comment]
@lily-mayfield seems to be in the postgress config or the link

**Comment by lily-mayfield** [comment]
@meyskens Did I nail all the changes in the new commits I made to PR #51 (except for `alpine`)? If so, I'd like to close this and perhaps let you open a PR in the future which implements `alpine`.

**Pull request closed without merging**